lte/gateway/c/core/oai/test/amf/test_amf_stateless.cpp

```diff
@@ -447,14 +655,39 @@ class AMFAppStatelessTest : public ::testing::Test {
 
     init_task_context(TASK_MAIN, nullptr, 0, NULL, &amf_app_task_zmq_ctx);
 
-    amf_app_desc_p = get_amf_nas_state(true);
+    amf_app_desc_p = get_amf_nas_state(false);
   }
 
   virtual void TearDown() {
     clear_amf_nas_state();
     clear_amf_config(&amf_config);
     destroy_task_context(&amf_app_task_zmq_ctx);
     itti_free_desc_threads();
+    AMFClientServicer::getInstance().map_table_key_proto_str.clear();
+    AMFClientServicer::getInstance().map_imsi_ue_proto_str.clear();
+  }
+
+  // This Function mocks mme restart.
+  void pseudo_amf_restart() {
+    vector<amf_ue_ngap_id_t> ue_id_vector;
+    for_each(ue_context_map.begin(), ue_context_map.end(), [&](auto& it) {
+      ue_id_vector.push_back(it.first);
+    });
+    for (auto& ue_id : ue_id_vector) {
+      ue_m5gmm_context_t* ue_context_p =
+          amf_ue_context_exists_amf_ue_ngap_id((amf_ue_ngap_id_t) ue_id);
+      if (ue_context_p) {
+        // Clean up the procedures
+        amf_nas_proc_clean_up(ue_context_p);
+        delete ue_context_p;
```

Comment:
Use [unordered_map::erase](https://www.cplusplus.com/reference/unordered_map/unordered_map/erase/) to delete `ue_context_p` as you iterate over the map.

Reply:
Comment Addressed